news/_posts/2019-12-25-prestashop-in-2019-and-beyond-part-3-the-future-architecture.md

```diff
@@ -0,0 +1,269 @@
+
+But as a code archeologist, I can't say I was surprised. After all, where was the master plan, the big picture, any _guidance_ to return strayed developers back to the right path? I was unable to find any trace of such thing.
+
+Software engineering can be easily assimilated to traditional engineering. Imagine a house being built with no plan, without a precise idea of what you want it to look like in the end or how many rooms it's supposed to have. Now, picture an architect leading a construction site and going like this: "Okay guys, just pile up some bricks here and add some columns there and we'll see where it takes us." It sounds ludicrous, right? So why would it be acceptable to lead a development as complex as PrestaShop like that?
+
+**If you want to get somewhere, _anywhere_, the first thing you have to do is decide where you want to go.** It's only _after that_ that you can start thinking on how you'll get there. 
+
+So we started working on that.
+
+When I first started talking about a new architecture during my first presentation before agencies at PrestaShop's Paris HQ back in 2017, attendants were rolling their eyes. "_Seriously?_", they said, "Are you seriously changing everything again? We're already struggling with 1.7 and now we'll have to spend money rebuilding our themes and modules _again_?". 
+
+I already knew by then that an architecture change isn't done overnight—especially for a development platform. So I reassured them by saying that it was our long-term vision and that they didn't have to worry about that right now, and that was the end of it.
+
+But even today, [four years after announcing 1.7 would be Symfony-based][introducing-symfony], many people are convinced that the Symfony migration was a waste of time and money, and that PrestaShop just did it to "force people into buying new modules". Some think that Symfony is a solution looking a problem to fix, a problem that didn't exist in the first place. I hope that after having read the previous two parts, you already understand that we're not doing it either, as someone once put it, "just because we enjoy beautiful code" (even if we, as most developers, actually _do_ enjoy beautiful code). There are many issues to fix,   yes, I know. But migrating to Symfony is a way of progressively giving more sense to PrestaShop's code, using modern tools, and improve its quality through incremental refactoring.
```

Comment:
"solution looking a problem to fix"
a for is missing

Reply:
You can't use "either" alone like that. You need a "or" a neither somewhere.
I would simply remove it

Reply:
> a for is missing

Whoops

> You can't use "either" alone like that. 

Yes it's correct using it that way, it works like the French "non plus". But it was confusing in this context so I'm removing it.

How about this?

```suggestion
But even today, [four years after announcing 1.7 would be Symfony-based][introducing-symfony], many people are convinced that the Symfony migration was a waste of time and money, and that PrestaShop just did it to "force people into buying new modules". Some think that Symfony is a solution looking for a problem to fix, a problem that didn't exist in the first place. I hope that after having read the previous two parts, you already understand that we're not doing it, as someone once put it, "just because we enjoy beautiful code" (even if we, as most developers, actually _do_ enjoy beautiful code). There are many issues to fix, yes, I know. But migrating to Symfony is a way of progressively giving more sense to PrestaShop's code, using modern tools, and improve its quality through incremental refactoring.
```